Vistara : Mailbox command to read coredump info. from qspi

manoranjan-lab opened the pull request:
Summary:
Mailbox command to Read coredump information from qspi, to a file in host machine over CXL interface

Test Plan:
$ cxl get-coredump mem0
mem0: firmware status: 5
Stored coredump verification failed or there is no stored coredump.
$ cxl trigger-coredump mem0
$ cxl get-coredump

 usage: cxl get-coredump <mem0> [<mem1>..<memN>] [<options>]

.    -v, --verbose         turn on debug

$ cxl get-coredump mem0

 Successfully collected coredump (size 358839 Bytes) at /tmp/coredump_mem0.bin

Note:
Provide mem1 to get mem1 coredump in place of mem0 for the above commands

venkateshmeta commented on cxl/lib/libcxl.c:
```diff
@@ -13319,6 +13319,64 @@ CXL_EXPORT int cxl_memdev_trigger_coredump(struct cxl_memdev *memdev)
+    FILE *fp;
+    const char *devname = cxl_memdev_get_devname(memdev);
+    char coredump_file[COREDUMP_FILE_NAME_LEN] = {0,};
+    sprintf(coredump_file,"%s_%s.bin",COREDUMP_FILE_NAME,devname);
```
add small comment to indicate coredump postfixed with devname

manoranjan-lab replied:
addressed

venkateshmeta commented on cxl/lib/libcxl.c:
```diff
@@ -13319,6 +13319,64 @@ CXL_EXPORT int cxl_memdev_trigger_coredump(struct cxl_memdev *memdev)
+    char coredump_file[COREDUMP_FILE_NAME_LEN] = {0,};
+    sprintf(coredump_file,"%s_%s.bin",COREDUMP_FILE_NAME,devname);
+    fp = fopen(coredump_file, "wb");
+     if (fp == NULL) {
```
check this identation deviation

manoranjan-lab replied:
taken care

venkateshmeta commented on cxl/lib/libcxl.c:
```diff
@@ -13319,6 +13319,64 @@ CXL_EXPORT int cxl_memdev_trigger_coredump(struct cxl_memdev *memdev)
+        fprintf(stderr,"Error opening the %s file.\n", coredump_file);
+        return -ENOENT; // Indicate an error
+    }
+     do {
```
identation deviation.

manoranjan-lab replied:
taken care

venkateshmeta commented on cxl/lib/libcxl.c:
```diff
@@ -13319,6 +13319,64 @@ CXL_EXPORT int cxl_memdev_trigger_coredump(struct cxl_memdev *memdev)
+    rc = cxl_cmd_get_mbox_status(cmd);
+    if (rc != 0) {
+        fprintf(stderr, "%s: firmware status: %d\n", cxl_memdev_get_devname(memdev), rc);
+         fprintf(stderr,"%s\n", "Stored coredump verification failed or there is no stored coredump.");
```
indentation

manoranjan-lab replied:
taken care

venkateshmeta commented on cxl/lib/libcxl.c:
```diff
@@ -13319,6 +13319,64 @@ CXL_EXPORT int cxl_memdev_trigger_coredump(struct cxl_memdev *memdev)
+        goto out;
+    }
+    if (cmd->send_cmd->id != CXL_MEM_COMMAND_ID_GET_COREDUMP) {
+                fprintf(stderr, "%s: invalid command id 0x%x (expecting 0x%x)\n",
```
indentation

manoranjan-lab replied:
taken care

venkateshmeta commented on cxl/lib/libcxl.c:
```diff
@@ -13319,6 +13319,64 @@ CXL_EXPORT int cxl_memdev_trigger_coredump(struct cxl_memdev *memdev)
+                                CXL_MEM_COMMAND_ID_GET_COREDUMP);
+                return -EINVAL;
+        }
+    bytes_read += cmd->send_cmd->out.size;
```
+ identaion

manoranjan-lab replied:
taken care

venkateshmeta commented on cxl/lib/libcxl.c:
```diff
@@ -13319,6 +13319,64 @@ CXL_EXPORT int cxl_memdev_trigger_coredump(struct cxl_memdev *memdev)
+     fprintf(stdout, "\nSuccessfully coredump info collected(size %d Bytes) at %s \n",
+                     bytes_read, coredump_file);
+out:
+        fclose(fp);
```
please once cross check all places where indentation deviates.

manoranjan-lab replied:
taken care

venkateshmeta commented:
could you please update testplan properly for help command which is showing as below not completely.
2> cxl get-coredump
usage: cxl get-coredump [..] []

  -v, --verbose         turn on debug

balajhi-dn commented:
Please update Title, Summary and Test Plan

manoranjan-lab committed:
VISTARA : Mailbox command to read coredump info. from qspi

Summary:
	Mailbox command to Read coredump info. from qspi to a file in host machine over CXL interface

Test Plan:
  cxl get-coredump mem0 / mem1

  cxl trigger-coredump mem0
  cxl trigger-coredump mem1

  cxl get-coredump

Usage: cxl get-coredump <mem0> [<mem1>..<memN>] [<options>]

    -v, --verbose         turn on debug

Output:
	> TO CHECK IF THE COREDUMP IS GENERATED
	ls -lrt /tmp

Reviewers:

Subscribers:

Tasks:

Tags:

Signed-off-by: Manoranjan Prusty <prustmanoranjan@meta.com>

venkateshmeta reviewed (approved):
LGTM

balajhi-dn reviewed (approved):
LGTM
